docs/application/web/guides/machine-learning/single.md

```diff
@@ -0,0 +1,160 @@
+
+- Invoking the neural network model
+
+  After creating the SingleShot instance, you can [invoke the model](#invoke) with the input data and get the inference output result.
+
+- Fetching the inference result
+
+  You can [fetch the inference result](#fetch-the-result) after invoking the respective model.
+
+## Prerequisites
+
+To access files using the MLSingleShot API (in [mobile](../../api/latest/device_api/mobile/tizen/ml_single.html), [wearable](../../api/latest/device_api/wearable/tizen/ml_single.html), and [TV](../../api/latest/device_api/tv/tizen/ml_single.html) applications), the application has to request storage privileges by adding the following privileges to the `config.xml` file:
+
+```xml
+<!-- for accessing internal storage only -->
+<tizen:privilege name="http://tizen.org/privilege/mediastorage"/>
+<!-- for accessing external storage only -->
+<tizen:privilege name="http://tizen.org/privilege/externalstorage"/>
+```
+
+Additionally, to access files using the MLSingleShot API (in [mobile](../../api/latest/device_api/mobile/tizen/ml_single.html) and [wearable](../../api/latest/device_api/wearable/tizen/ml_single.html) applications), the application has to request [proper storage permissions](../security/privacy-related-permissions.md) using the PPM API (in [mobile](../../api/latest/device_api/mobile/tizen/ppm.html) and [wearable](../../api/latest/device_api/wearable/tizen/ppm.html) applications).
+
+## Open the model
+
+1. To load the model from storage, use `openModel` method:
+```javascript
+  var model = tizen.ml.single.openModel("documents/mobilenet_v1_1.0_224_quant.tflite");
+```
+
+2. If the model supports dynamic tensors, you can:
+
+- provide custom input and output tensors:
+```javascript
+  var input = new tizen.ml.TensorsInfo()
+  input.addTensorInfo("input", "FLOAT32", [3])
+  var output = new tizen.ml.TensorsInfo()
+  output.addTensorInfo("output", "FLOAT32", [3])
+  var model = tizen.ml.single.openModel("documents/dynamic.tflite", input, output);
+  input.dispose();
+  output.dispose();
+```
+
+- set the model's `input` tensor:
+```javascript
+  var input = new tizen.ml.TensorsInfo()
+  input.addTensorInfo("input", "FLOAT32", [3])
+  model.input = input;
+  input.dispose();
+```
+
+- open model with dynamic mode, which allows you to [invoke mode](#invoke) with variable tensors.
+
+```javascript
+  var model = tizen.ml.single.openModel("documents/mobilenet_v1_1.0_224_quant.tflite", null, null, "ANY", "ANY", true);
+```
+
+3. You can also provide parameters that will define framework and hardware for your model:
+
+```javascript
+  var model = tizen.ml.single.openModel("documents/mobilenet_v1_1.0_224_quant.tflite", null, null, "TENSORFLOW_LITE", "ANY");
+```
+
+4. Remember to always close SingleShot objects when you don't need them anymore:
+
+```javascript
+  model.close()
+```
+
+5. You can also open model asyncronously with `openModelAsync` method.
```

Comment:
maybe add a typical usecase? e.g. to avoid blocking UI waiting for reading big models?